src/nomad_material_processing/__init__.py

```diff
@@ -49,287 +52,364 @@
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section()
+    lab_id = Quantity(
+        type=str,
+        a_eln=ELNAnnotation(
+            component="StringEditQuantity",
+        ),
+    )
+    reference = Quantity(
+        type=ThinFilm,
+        a_eln=ELNAnnotation(
+            component="ReferenceEditQuantity",
+        ),
+    )
+
+
+class SubstrateReference(CompositeSystemReference):
+    """
+    A section for describing a system component and its role in a composite system.
+    """
+
+    lab_id = Quantity(
+        type=str,
+        a_eln=ELNAnnotation(
+            component="StringEditQuantity",
+        ),
+    )
+    reference = Quantity(
+        type=SubstrateSlab,
+        a_eln=ELNAnnotation(
+            component="ReferenceEditQuantity",
+        ),
     )
 
 
 class ThinFilmStack(CompositeSystem):
-    '''
+    """
     A stack of `ThinFilm`(s). Typically deposited on a `Substrate`.
-    '''
+    """
+
     m_def = Section()
-    substrate = Quantity(
-        type=Substrate,
-        description='''
+    # layers = Quantity(
+    #     type=ThinFilm,
+    #     description='''
+    #     An ordered list (starting at the substrate) of the thin films making up the
+    #     thin film stacks.
+    #     ''',
+    #     a_eln=ELNAnnotation(
+    #         component=ELNComponentEnum.ReferenceEditQuantity,
+    #     ),
+    #     shape=["*"],
+    # )
+    layers = SubSection(
+        description="""
+        An ordered list (starting at the substrate) of the thin films making up the
+        thin film stacks.
+        """,
+        section_def=ThinFilmReference,
+        repeats=True,
+    )
+    substrate = SubSection(
+        description="""
         The substrate which the thin film layers of the thin film stack are deposited
         on.
-        ''',
+        """,
+        section_def=SubstrateReference,
+        repeats=True,
+    )
+    ####################################################################################
+
+
+########  !!! Here I see layers and substarte. What about components subsection present in CompositeSystem class?  ####TODO
+####################################################################################
+
+
+class ThinFilmStackReference(CompositeSystemReference):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section()
+    lab_id = Quantity(
+        type=str,
         a_eln=ELNAnnotation(
-            component=ELNComponentEnum.ReferenceEditQuantity,
+            component="StringEditQuantity",
         ),
     )
-    layers = Quantity(
-        type=ThinFilm,
-        description='''
-        An ordered list (starting at the substrate) of the thin films making up the
-        thin film stacks.
-        ''',
+    reference = Quantity(
+        type=ThinFilmStack,
         a_eln=ELNAnnotation(
-            component=ELNComponentEnum.ReferenceEditQuantity,
+            component="ReferenceEditQuantity",
```

Comment:
What's the reason for removing the `ELNComponentEnum`? I would suggest we rather change all of them to use this.

Reply:
this is just a mismatching of the changes in github. there are 80 lines of difference and the lines belong to different classes.

Also, I simply forget to place it as you did elsewhere, I will

Reply:
if you find it somewhere else, feel free to chnage it when you will start committing here

Reply:
Okay, thanks for clarifying. Will do!